src/main/java/duke/utility/parser/Parser.java

```diff
@@ -0,0 +1,256 @@
+package duke.utility.parser;
+
+import duke.duke_exception.DukeException;
+import duke.tasklist.TaskList;
+import duke.tasklist.task_types.Deadline;
+import duke.tasklist.task_types.Event;
+import duke.tasklist.task_types.Task;
+import duke.tasklist.task_types.ToDo;
+import duke.utility.ui.UiMessage;
+
+
+/**
+ * Represents a <code>Parser</code> object that perfroms the necessary operations based on the user
+ * input.
+ * 
+ * @author Brian Quek
+ */
+public class Parser {
+    private static DukeException wrongCommandFormat =
+            new DukeException("Wrong command format inserted.");
+    private static DukeException noNumericParam =
+            new DukeException("Parameter is not a numerical value.");
+    private static DukeException noSpecialParam =
+            new DukeException("Missing special param e.g /by, /from, /to");
+    private static DukeException emptyParam = new DukeException("Empty parameter inserted.");
+
```

Comment:
Consider leaving only a single line separator between attributes and methods instead of 2 lines to ensure consistency across code files.